sbin/spark-daemon.sh

```diff
@@ -164,7 +164,7 @@ run_command() {
   echo "$newpid" > "$pid"
   sleep 2
   # Check if the process has died; in that case we'll tail the log so the user can see
-  if [[ ! $(ps -p "$newpid" -o comm=) =~ "java" ]]; then
+  if [[ -z $(ps --no-headers -p "$newpid") ]]; then
```

Comment:
I don't see the option `--no-headers` on OS X, so I suspect it could be a BSD vs Gnu thing. It will fail though.

I don't see how this change addresses the problem though. It still occurs 2 seconds after the start, but it's not checking whether the java process started. Do you mean some kind of loop here, like checking 3 times?

What's the scenario where the bash script takes a long time to run?